checked_message.go

```diff
@@ -20,7 +20,13 @@
 
 package zap
 
-import "github.com/uber-go/atomic"
+import (
+	"sync"
+
+	"github.com/uber-go/atomic"
+)
+
+var cmPool sync.Pool
```

Comment:
I think that the code below would be a little clearer if you defined a `New` method on the pool that returned `&CheckedMessage{}`.

Also, please underscore-prefix global vars (per the in-house style guide - I've probably been a bit inconsistent in zap, but I'll clean up eventually).